WebApp/src/test/java/ev/projects/webapp/controllers/CaseControllerTest.java

```diff
@@ -1,93 +1,158 @@
-import java.util.Date;
 import java.util.List;
 import java.util.Optional;
 
+import static ev.projects.webapp.utils.EntitiesCreator.createCase;
 import static org.junit.jupiter.api.Assertions.*;
 
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 @SpringBootTest(classes = WebAppApplication.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 public class CaseControllerTest {
 
     @LocalServerPort
     private int port;
 
     private String baseUrl;
-    @Value("${login.username}")
-    private String username;
-    @Value("${login.password}")
-    private String password;
 
-    Field privateField = CaseRepository.class.getDeclaredField("cases");
+    private IUserRepository userRepository;
+    private UserService userService;
+    private ICaseRepository caseRepository;
+
+    String username1 = "Smith";
+    String username2 = "Jane";
+    String password1 = "pass";
+    String password2 = "smith";
+
+    private long userCountOffset;
+    private long caseCountOffset;
 
-    public CaseControllerTest() throws NoSuchFieldException {
+    @BeforeAll
+    public void startUp() {
+        loadData();
+    }
+
+    @Autowired
+    public CaseControllerTest(IUserRepository userRepository, UserService userService, ICaseRepository caseRepository) {
+        this.userRepository = userRepository;
+        this.userService = userService;
+        this.caseRepository = caseRepository;
+    }
+
+    public void loadData() {
+        userCountOffset = userService.add(new User(username1, password1)).getID() - 1;
+        userService.add(new User(username2, password2));
+        caseCountOffset = caseRepository.save(createCase("English", "English lessons",
+                userService.getByUsername(username1).get())).getID() - 1;
+        caseRepository.save(createCase("Math", "Math horrors",
+                userService.getByUsername(username2).get()));
     }
 
     @BeforeEach
     public void start() {
-        privateField.setAccessible(true);
-        baseUrl = "http://localhost:" + port;
+        baseUrl = "http://localhost:" + port + "/api/cases/";
     }
 
     @Test
-    public void getCasesTest() throws IllegalAccessException {
-        List<Case> cases = MockData.generateCases();
-        privateField.set(null, cases);
-        String url = baseUrl + "/api/cases/";
-        TestRestTemplate restTemplate = new TestRestTemplate().withBasicAuth(username, password);
-        ResponseEntity<List<Case>> casesResponse = restTemplate.exchange(url, HttpMethod.GET,
-                null, new ParameterizedTypeReference<List<Case>>() {});
-        List<Case> responseCases = casesResponse.getBody();
-        assertNotNull(responseCases);
-        assertEquals(2, responseCases.size());
-        Case aCase = responseCases.get(1);
-        assertNotNull(aCase);
-        assertEquals(aCase.getID(), 2);
-        assertEquals(aCase.getTitle(), "Math");
+    public void getCasesTest() {
```

Comment:
Name of unit test should be readable and describe it's purpose.
E.g. caseCasesMethodShouldReturnAllCases

Reply:
Will rename it. It was named that way to notify that it tests the controller method of the same name ("getCases").

Reply:
Renamed at 27f8c4cfed5161b850651143a5af42c7480f769d. Is it clear enough?

Reply:
it is not a big deal, but will take some time for other developers to understand the purpose of the test.